airflow/providers/amazon/aws/hooks/emr_containers.py

```diff
@@ -123,7 +123,9 @@ def get_job_failure_reason(self, job_id: str) -> Optional[str]:
                 virtualClusterId=self.virtual_cluster_id,
                 id=job_id,
             )
-            reason = response['jobRun']['failureReason']
+            failure_reason = response['jobRun']['failureReason']
+            state_details = response["jobRun"]["stateDetails"]
+            reason = f"{failure_reason} - {state_details}"
```

Comment:
Is it backward compatible?

Should we also modify the test to include the `stateDetails`?
https://github.com/apache/airflow/blob/890bd4310e12a0a4fadfaec1f9b36d2aaae6119e/tests/providers/amazon/aws/operators/test_emr_containers.py#L94-L103

Reply:
It is backwards compatible.
Thanks, I added `stateDetails` to return_value.